src/commands/new/glee.ts

```diff
@@ -0,0 +1,42 @@
+import {Flags} from '@oclif/core';
+import { fstat, promises as fPromises } from 'fs';
+import Command from '../../base';
+import { resolve } from 'path';
+import path from 'path';
+import fs from 'fs-extra';
+
+const { writeFile, readFile } = fPromises;
+const fs = require('fs-extra');
+
+export default class NewGlee extends Command {
+  static description = 'Creates a new Glee project';
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    name: Flags.string({ char: 'n', description: 'name of the project', default: 'glee-project' }),
+  };
+
+  async run() {
+    const { flags } = await this.parse(NewGlee); // NOSONAR
+
+    const projectName = flags.name;
+
+    const PROJECT_DIRECTORY = path.join(process.cwd(), projectName); // cwd-> /Users/pedro.ramos/cli
+    const GLEE_TEMPLATES_DIRECTORY = resolve(__dirname, '../../../create-glee-app/templates/default'); // __dirname-> /Users/pedro.ramos/cli/lib/commands/new
+
+    try {
+      await fPromises.mkdir(PROJECT_DIRECTORY);
+      console.log('Glee project folder created!');
+    } catch (err) {
+      console.error('An existing folder is already using the same name of your Glee project. Please specify a different name for the new project, or just remove the existing folder from the current directory.');
```

Comment:
Suggesting a better message error IMHO:

```suggestion
      console.error(`Unable to create the project. We tried to use "${projectName}" as the directory of your new project but it already exists (${PROJECT_DIRECTORY}). Please specify a different name for the new project. For example, run the following command instead:\n\nasyncapi new glee --name ${projectName}-1`);
```

Reply:
I guess I should suggest here `asyncapi new glee --name ${projectName}-1` or `asyncapi new project --name ${projectName}-1` depending on which subcommand has been used ('glee' or 'project'). My question is: how could I access to that value so I can use it in the code? 🤔

Reply:
Yeah, there are many ways but since you're using classes I think you can take advantage of class inheritance. It's hard to explain here but I'll try. On the project.ts file you do something like this:

```ts
import NewGleeCommand from './glee'

export default class NewProjectCommand extends NewGleeCommand {
  constructor(...args) {
    super(..args)
    this.commandName = 'project'
  }
}
```

and on the glee.ts file you put:

```ts
export default class NewGleeCommand extends Command {
  protected commandName = 'glee'
```

Then on each of the places where you need to retrieve that name, you do `this.commandName` and that will give you either `proejct` or `glee`, depending on the command that was used.

It's hard to explain here but I'm happy to jump into a call with you tomorrow if you need it.